src/main/java/seedu/address/storage/JsonAdaptedRecurringSession.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.session.Duration;
+import seedu.address.model.session.Fee;
+import seedu.address.model.session.Interval;
+import seedu.address.model.session.RecurringSession;
+import seedu.address.model.session.Session;
+import seedu.address.model.session.SessionDate;
+import seedu.address.model.session.Subject;
+
+public class JsonAdaptedRecurringSession extends JsonAdaptedSession {
+    private String interval;
+    private String lastSessionDate;
+
+    /**
+     * JsonCreator for Recurring Session
+     * @param sessionDate
+     * @param duration
+     * @param subject
+     * @param fee
+     * @param interval
+     * @param lastSessionDate
+     */
+    @JsonCreator
+    public JsonAdaptedRecurringSession(@JsonProperty("sessionDate") String sessionDate,
```

Comment:
Hi, so sorry, but I am not sure where the error is.

However, if you run the command:
add_rec_session n/Alex Yeoh d/2020-03-03 t/12:00 k/90 s/Math f/40 b/7 e/2020-03-17

then close the application, open the application again.

Run the command:
delete_session n/Alex Yeoh i/<number of the recurring session>
You will notice that the output will be "Deleted Session: Date: 2020-03-03; Time: 12:00; Duration: 90; Subject: Math; Fee: 40.0" instead of "Deleted Recurring Session: Date: 2020-03-03; Time: 12:00; Duration: 90; Subject: Math; Fee: 40.0; Interval: 7; Last Date: 2020-03-17".

Hence, the although the storage is storing the interval and last session value inside the json file, I think when the system reads from the storage, it creates a session instead of a recurring session. Might need to look into that.

Reply:
Ok found it will fix

Reply:
actually no idea

Reply:
Issue is solved by adding type information in json, as necessary for deserialisation polymorphism.